### DISCUSSION TITLE
Welcome to tetrad Discussions!

### DISCUSSION THREAD

**Discussion by jdramsey** [cmu-phil/tetrad#1677] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jdramsey** [comment]
Hi, just turning on this discussion board (sorry I just saw it!). I'll follow any comments that are made, so please ask away!

I do a lot of the programming on this project and am happy to answer questions.

**Comment by cg09** [comment]
??? Where is it?

**Comment by kvb2univpitt** [comment]
Nice!

**Comment by jdramsey** [comment]
Oh, the link is down below:

> Reply to this email directly, view it on GitHub
> <https://github.com/cmu-phil/tetrad/discussions/1677>, or unsubscribe
> <
https://github.com/notifications/unsubscribe-auth/AD4Y3OM2KUXALRRLKDCSDZDXVRQOFANCNFSM6AAAAAA3SAU464
>
> .

:)

**Comment by knoolee** [comment]
Hello,

I am currently facing difficulties loading the .tet file compiled in TETRAD 6.7.0 when using the newer version of TETRAD 7.6.1. Additionally, when attempting to open the older version of TETRAD 6.7.0, I observed that the GUI does not appear, along with the notification I've included below.

![download](https://github.com/cmu-phil/tetrad/assets/50002741/f7c08fb1-05bd-437f-8775-2a318be96bb9)

Thank you,
Knoo

**Reply by jdramsey** [reply]
Oh boy--6.7.0-0 is far too old to support; there have been too many changes. You can see the release history on the GitHub page,

https://github.com/cmu-phil/tetrad

Also, we plan to release version 7.6.2 next Wednesday, 2024-1-31, we  with these revisions:

https://github.com/cmu-phil/tetrad/wiki/Forthcoming-fixes

Could you reconstruct the session using 7.6.1? Do you have the datasets you used? I could help; maybe there are better ways to do it now.

**Reply by knoolee** [reply]
Thank you for your response. Rebuilding the analysis with the new version seems feasible, but the inability to access the old compiled tetrad is a challenge I would like to address mainly having access would mean having access to past histories of analyses. However, I understand if this is not a primary focus for the TETRAD development team.

**Reply by jdramsey** [reply]
The main obstacle is that the versions of the Java JDK those older versions of Tetrad used aren't even available anymore, and they won't run under current versions of the JDK.  If you happen to have one of the older JDKs on your machine you should be able to run it, but there's no way to get those online anymore, to the best of my knowledge. That was even before JDK 1.8, and you'd need a pretty old version of JDK 1.8 if you can even get it to run with that.

Do you have the JDKs that you used to run 6.7.0-0 on?

**Reply by jdramsey** [reply]
For example, for Amazon's Corretto JDK, the available options are here:

https://aws.amazon.com/corretto/?filtered-posts.sort-by=item.additionalFields.createdDate&filtered-posts.sort-order=desc

The version of 1.8 given here is a recent version of 1.8, though they recommend not using it, as it's incompatible with later versions of the JDK.

**Comment by sajadabvi** [comment]
Hello. I am trying to run FASK and Alasso in IDE environment. Can you please point me to an example script to see how I can use these function and classes?

thank you.

**Reply by jdramsey** [reply]
What language are you using? You could use FASK from Java, but it is also possible to access it from Python or R.

**Reply by sajadabvi** [reply]
Thank you so much for your prompt reply. I prefer python, but Java is good too. I have looked at https://github.com/cmu-phil/py-tetrad/tree/main , but could not find an example script. 

Thank you.

**Reply by jdramsey** [reply]
Oh, good point. I did not put FASK in the py-tetrad scripts. I will add it and get back to you. Once you see how to use it from py-tetrad you'll see how to use it from Java.

**Reply by jdramsey** [reply]
OK, I've added it to TetradSearch.py and added an example to run_continuous.py in py-tetrad. The example is silly, just to show you how to work it. The default parameters I used were ones used in our paper where we introduced FASK--these:

![image](https://github.com/cmu-phil/tetrad/assets/9853255/e392b683-a92d-4cae-b8bf-201129596572)

This paper:

Sanchez-Romero, R., Ramsey, J. D., Zhang, K., Glymour, M. R., Huang, B., & Glymour, C. (2019). Estimating feedforward and feedback effective connections from fMRI time series: Assessments of statistical methods. Network Neuroscience, 3(2), 274-306.

Let me know if that's sufficient to get you started.

**Reply by sajadabvi** [reply]
Thank you very much. Are you referring to the files in this repo?
https://github.com/cmu-phil/py-tetrad/blob/main/pytetrad/run_continuous.py
and 
https://github.com/cmu-phil/py-tetrad/blob/main/pytetrad/tools/TetradSearch.py

seems like they have not been updated in the last two months. I can't see you last week's commit and update

**Comment by jdramsey** [comment]
My solution above should make it easy to run FASK, though you may want to delve into more detailed JPype coding. Let me know. I can also write this out using JPype, which is closer to Java. Writing it in Java is not hard; I could give you some pointers.

**Comment by zhangdai-star2945** [comment]
I am reaching out to inquire about the process for translating the software interface into Chinese. Currently, I have successfully translated the HTML help file, but the main interface remains in English. This poses a challenge for local students who would benefit from a fully localized version. Could you provide guidance or tools that support interface translation?

**Comment by jdramsey** [comment]
Thanks for letting me know; I'll try pushing them again.

Joe

**Comment by jdramsey** [comment]
They're there now; check it out to make sure. I'm not sure why they didn't
push last time.

Best,

Joe

**Reply by sajadabvi** [reply]
Its working. thank you very much. I want to run the following methods:
![Screenshot 2024-06-19 at 7 25 40 PM](https://github.com/cmu-phil/tetrad/assets/61247228/79ea7b35-13c8-42dd-a648-713dd39b7bc6)

I believe I can do the highlighted ones using the key:left_right_rule = 1,2,... in run_fask(). Can you please point me to the function for the other methods?

Thank you,
Sajad

**Reply by jdramsey** [reply]
Ah, I knew someone was asking about FASK, but I lost track of where! Sorry. I'm glad FASK is working for you now. You know, you're the first person to ask about FAS + Skew or RSkew in a very long time; I had taken that out of Tetrad. I suppose I could give you an algorithms to work with though if it would be helpful. We have one currently in Tetrad called BOSS-LiNGAM, which starts with a BOSS CPDAG and orients the undirected edges using the FASK left-right rule. That could be generalized. Or I could just re-add an algorithm that uses FAS adjacencies and lets you orient all edges pairwise. 

Can you tell me what you goal is? Maybe that would help me to see what to do.

**Reply by sajadabvi** [reply]
I am comparing my method with some of the previous methods for causal discovery from time series. I would appreciate if you can show me how to run these algorithms. additionally, if you have new and better methods, please let me know as well.
